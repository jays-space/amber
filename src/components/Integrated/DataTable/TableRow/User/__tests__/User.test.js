import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { User } from "..";

describe("<User />", () => {
  it("should render correctly", () => {
    const candidate = false;

    render(
      <User
        name="john doe"
        avatar="https://randomuser.me/api/portraits/women/92.jpg"
        candidate={candidate}
      />
    );

    const userComponent = screen.getByTitle("user-details");
    expect(userComponent).toBeInTheDocument();
    expect(userComponent).toMatchSnapshot();
  });

  it("should render candidate correctly", () => {
    const candidate = true;

    render(
      <User
        name="john doe"
        avatar="https://randomuser.me/api/portraits/women/92.jpg"
        candidate={candidate}
      />
    );

    const userComponent = screen.getByTitle("user-details");
    expect(userComponent).toBeInTheDocument();
  });

  it("should render default correctly", () => {
    render(
      <User
        name="john doe"
        avatar="https://randomuser.me/api/portraits/women/92.jpg"
      />
    );

    const userComponent = screen.getByTitle("user-details");
    expect(userComponent).toBeInTheDocument();
  });

  it("should render user avatar", () => {
    render(
      <User
        name="john doe"
        avatar="https://randomuser.me/api/portraits/women/92.jpg"
      />
    );

    const avatar = screen.getByRole("img");
    expect(avatar).toBeInTheDocument();
  });
});
