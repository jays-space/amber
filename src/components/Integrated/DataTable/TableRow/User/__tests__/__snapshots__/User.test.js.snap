// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render correctly 1`] = `
<div
  class="flex items-center"
  title="user-details"
>
  <img
    alt="john doe"
    class="w-16 lg:w-10 mr-4 aspect-square rounded-full hidden lg:flex"
    src="https://randomuser.me/api/portraits/women/92.jpg"
  />
  <div
    class="flex flex-col overflow-hidden"
  >
    <p
      class="tracking-wide  text-slate-700 flex lg:hidden capitalize"
      data-testid="paragraph"
    >
      owner
    </p>
    <p
      class="tracking-wide  text-cyan-900 font-semibold pr-4 text-sm md:text-lg capitalize whitespace-nowrap text-ellipsis overflow-hidden"
      data-testid="paragraph"
    >
      john doe
    </p>
  </div>
</div>
`;
